### PR TITLE
update otc for v4 compatibility

### PR DESCRIPTION
**Pull request by dmosites**

### CODE REVIEW

File: src/features/orders/ordersActions.ts [hunk 1]
```diff
@@ -277,6 +277,19 @@ export const approve =
...
 
+      if (isAppError(tx)) {
+        const appError = tx;
+        dispatch(setErrors([appError]));
+
+        if (appError.error && "message" in appError.error) {
+          dispatch(declineTransaction(appError.error.message));
+        }
+
+        dispatch(setStatus("failed"));
```
Comment (piersss) [inline src/features/orders/ordersActions.ts:288]:
```suggestion
        dispatch(setStatus("failed"));
        handleOrderError(dispatch, tx);
```

This does the same basically.

File: src/features/orders/ordersHelpers.ts [hunk 2]
```diff
@@ -203,7 +209,7 @@ export async function check(
   provider: ethers.providers.Web3Provider,
   isSwapWithWrap?: boolean
 ): Promise<AppError[]> {
-  const strings = await (
+  const [count, strings] = await (
```
Comment (piersss) [inline src/features/orders/ordersHelpers.ts:212]: strings is now undefined so it breaks here

File: src/features/orders/ordersActions.ts (hunk 1 above)
Comment (dmosites) [reply src/features/orders/ordersActions.ts:288]: Great. Had just copied this over from the function below, will update.

File: src/features/orders/ordersHelpers.ts (hunk 2 above)
Comment (dmosites) [reply src/features/orders/ordersHelpers.ts:212]: Works on my end and the swapErc20 contractV4Abi ABI indicates uint256, bytes32[]:
```
name: "check",
outputs: [
  { internalType: "uint256", name: "", type: "uint256" },
  { internalType: "bytes32[]", name: "", type: "bytes32[]" },
],
```

**Review by piersss — APPROVED** [review]
LGTM

**Pull request merged**